pkg/virt-launcher/notify-client/client.go

```diff
@@ -247,38 +247,50 @@ func eventCallback(c cli.Connection, domain *api.Domain, libvirtEvent libvirtEve
 		domain.ObjectMeta.DeletionTimestamp = &now
 		watchEvent := watch.Event{Type: watch.Modified, Object: domain}
 		client.SendDomainEvent(watchEvent)
-		events <- watchEvent
+		updateEvents(watchEvent, domain, events)
 	default:
 		if libvirtEvent.Event != nil {
 			if libvirtEvent.Event.Event == libvirt.DOMAIN_EVENT_DEFINED && libvirt.DomainEventDefinedDetailType(libvirtEvent.Event.Detail) == libvirt.DOMAIN_EVENT_DEFINED_ADDED {
 				event := watch.Event{Type: watch.Added, Object: domain}
 				client.SendDomainEvent(event)
-				events <- event
+				updateEvents(event, domain, events)
 			} else if libvirtEvent.Event.Event == libvirt.DOMAIN_EVENT_STARTED && libvirt.DomainEventStartedDetailType(libvirtEvent.Event.Detail) == libvirt.DOMAIN_EVENT_STARTED_MIGRATED {
 				event := watch.Event{Type: watch.Added, Object: domain}
 				client.SendDomainEvent(event)
-				events <- event
+				updateEvents(event, domain, events)
 			}
 		}
-		log.Log.Infof("7) OSINFO IN EVENT CALLBACK: %v", osInfo)
 		if interfaceStatus != nil {
 			domain.Status.Interfaces = interfaceStatus
 		}
 		if osInfo != nil {
 			domain.Status.OSInfo = *osInfo
 		}
-		if interfaceStatus != nil || osInfo != nil {
-			event := watch.Event{Type: watch.Modified, Object: domain}
-			client.SendDomainEvent(event)
-			events <- event
-		}
+
 		err := client.SendDomainEvent(watch.Event{Type: watch.Modified, Object: domain})
 		if err != nil {
 			log.Log.Reason(err).Error("Could not send domain notify event.")
 		}
 	}
 }
 
+var updateEvents = updateEventsClosure()
+
+func updateEventsClosure() func(event watch.Event, domain *api.Domain, events chan watch.Event) {
+	var firstAddEvent bool = true
+	var firstDeleteEvent bool = true
```

Comment:
Is more idiomatic to do the following it will continue to live inside the closure also.
```golang
	firstAddEvent bool := true
	firstDeleteEvent bool := true
```
This code is not thread safe since the state (firstAddEvent and firstDeleteEvent) is global, is eventCallback called by multiple goroutines ?

Reply:
> Is more idiomatic to do the following it will continue to live inside the closure also.
> 
> ```go
> 	firstAddEvent := true
> 	firstDeleteEvent := true
> ```
> 
> This code is not thread safe since the state (firstAddEvent and firstDeleteEvent) is global, is eventCallback called by multiple goroutines ?

Just from one go routine, and in that context only, it doesnt spawn a new one, its serial

Reply:
ack we can close this.

Reply:
changed to
```
firstAddEvent := true
firstDeleteEvent := true
```